### PR TITLE
Update `GetDefaultOpenPageBehavior`

### PR DESCRIPTION
**Pull request by erri120**
Part of #946. This PR implements the flowchart we developed earlier and most of #946.

### CODE REVIEW

**Comment by Al12rs** [comment]
Closing and reopening to update target of the diff

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/IWorkspaceController.cs [hunk 1]
```diff
@@ -112,18 +112,14 @@ public interface IWorkspaceController
...
-    /// </summary>
-    public OpenPageBehavior GetDefaultOpenPageBehavior();
-
     /// <summary>
     /// Returns true if a new panel can be created, else false.
     /// </summary>
     /// <param name="workspaceId">Workspace identifier.</param>
     bool CanCreateNewPanel(WorkspaceId workspaceId);
+
+    public OpenPageBehavior GetDefaultOpenPageBehavior(
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/IWorkspaceController.cs:121]: Missing comment

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/OpenPageBehaviorSettings.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public static readonly OpenPageBehaviorSettings Default = new()
+    {
+        // Primary Inputs
+        { new NavigationInput(MouseButton.Left, KeyboardModifiers.None), OpenPageBehaviorType.NewTab },
+        { new NavigationInput(KeyboardKey.Enter, KeyboardModifiers.None), OpenPageBehaviorType.NewTab },
+
+        // Others
+        { new NavigationInput(MouseButton.Left, KeyboardModifiers.Control), OpenPageBehaviorType.ReplaceTab },
+        { new NavigationInput(MouseButton.Left, KeyboardModifiers.Shift), OpenPageBehaviorType.NewTab },
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/OpenPageBehaviorSettings.cs:22]: I'm guessing these are temporary? Should we ask design?
I was thinking that we probably want Middle Click, like in browsers.

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/OpenPageBehaviorSettings.cs:22]: I forgot the middle click, I'll add that.

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs [hunk 3]
```diff
@@ -313,13 +313,61 @@ public void SetIcon(IconValue? icon, WorkspaceId workspaceId, PanelId panelId, P
...
-    public OpenPageBehavior GetDefaultOpenPageBehavior()
+    /// <inheritdoc/>
+    public OpenPageBehavior GetDefaultOpenPageBehavior(
+        PageData requestedPage,
+        NavigationInput input,
+        Optional<PageIdBundle> optionalCurrentPage)
     {
-        // TODO: Query user settings for default behaviors for:
-        // specific pages, specific interaction (mouse button modifiers), source workspace (if popped out or not)
+        const OpenPageBehaviorType fallback = OpenPageBehaviorType.NewTab;
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:322]: Shouldn't the fallback for opening from outside a workspace be Replace the first tab in first panel for now?
That is what design wanted at least

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:322]: This fallback is used in `GetValueOrDefault` calls when we try to look up a value in a Dictionary. If the Dictionary is populated correctly, then we will never use this fallback.

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:322]: I think what you're talking about is the default global setting behavior for the left menu.

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs [hunk 4]
```diff
@@ -313,13 +313,61 @@ public void SetIcon(IconValue? icon, WorkspaceId workspaceId, PanelId panelId, P
         tabViewModel.Header.Icon = icon;
     }
 
-    public OpenPageBehavior GetDefaultOpenPageBehavior()
+    /// <inheritdoc/>
+    public OpenPageBehavior GetDefaultOpenPageBehavior(
```
[the remark this answers is not in the document]
Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:317]: Not part of this PR.

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs (hunk 3 above)
Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:322]: Right, yeah that should be Replace Tab for now (I don't particularly like it myself, but the idea is to try it and see)

File: src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/WorkspaceController/WorkspaceController.cs:317]: I agree, I'll open an issue for design. Not sure when to schedule it yet

**Review by Al12rs — APPROVED** [review]

**Comment by Al12rs** [comment]
I'm sorry I'm slipping, I meant to test this locally with the View Mod contents. 
I just checked the code and LoadoutGridViewModel is using a custom behavior instead of GetDefaultBehavior. 
Could you do another PR to fix that? Or I can add it to the list if you are busy

**Pull request merged**